tsfm_public/toolkit/get_model.py

```diff
@@ -93,28 +94,51 @@ def get_model(
+                    if prediction_length <= h:
+                        selected_prediction_length = h
+                        break
 
-            LOGGER.info(f"Selected prediction_length = {selected_prediction_length}")
+            LOGGER.info(f"Selected TTM `prediction_length` = {selected_prediction_length}")
 
-            if selected_prediction_length != prediction_length:
+            if selected_prediction_length > prediction_length:
                 prediction_filter_length = prediction_length
                 LOGGER.warning(
-                    f"Requested `prediction_length` ({prediction_length}) is not exactly equal to any of the available TTM prediction lengths.\n\
-                    Hence, TTM will forecast using the `prediction_filter_length` argument to provide the requested prediction length.\n\
-                    Supported context lengths (CL) and forecast/prediction lengths (FL) for Model Card: {model_path} are\n\
-                    {SUPPORTED_LENGTHS[model_path_type]}"
+                    f"Requested `prediction_length` ({prediction_length}) is not exactly equal to any of the available TTM prediction lengths. Hence, TTM will forecast using the `prediction_filter_length` argument to provide the requested prediction length. Supported context lengths (CL) and forecast/prediction lengths (FL) for Model Card: {model_path} are {SUPPORTED_LENGTHS[model_path_type]}"
                 )
 
+            # Choose closest context length
+            available_context_lens = sorted(SUPPORTED_LENGTHS[model_path_type]["CL"], reverse=True)
+            selected_context_length = None
+            for cl in available_context_lens:
+                if cl <= context_length:
+                    selected_context_length = cl
+                    if cl < context_length:
+                        LOGGER.warning(
+                            f"Selecting TTM context length ({selected_context_length}) < Requested context length ({context_length} since exact match was not found.)"
+                        )
+                    break
+            if selected_context_length is None:
+                if force_return:
+                    selected_context_length = available_context_lens[-1]
+                    LOGGER.warning(
+                        f"Requested context length is too short. Requested = {context_length}. Available lengths for model_type = {model_path_type} are: {available_context_lens}. Returning the shortest context length model possible since `force_return=True`. Data needs to be handled properly, and it can affect the performance!"
+                    )
+                else:
+                    raise ValueError(
+                        f"Requested context length is too short. Requested = {context_length}. Available lengths for model_type = {model_path_type} are: {available_context_lens}. To return the shortest context length model possible, set `force_return=True`."
+                    )
+
```

Comment:
Minor nit in the exception below. Can we use `ValueError("In the current implementation, set `freq_prefix_tuning` to None for automatic model selection accordingly.")`

Reply:
Done. Thanks.